contracts/src/v0.8/functions/tests/1_0_0/Setup.t.sol

```diff
@@ -218,184 +260,237 @@ contract FunctionsClientRequestSetup is FunctionsSubscriptionSetup {
+      _offchainMetadata[i] = new bytes(0); // No off-chain metadata
+    }
+    report = abi.encode(_requestIds, _results, _errors, _onchainMetadata, _offchainMetadata);
+
+    // Build report context
+    uint256 h = uint256(
+      keccak256(
+        abi.encode(
+          block.chainid,
+          address(s_functionsCoordinator),
+          1,
+          s_signers,
+          s_transmitters,
+          s_f,
+          s_onchainConfig,
+          s_offchainConfigVersion,
+          s_offchainConfig
+        )
+      )
     );
-    // totalCostJuels = costWithoutCallbackJuels + adminFee + callbackGasCostJuels
-    s_fulfillmentCoordinatorBalance = totalCostJuels - s_adminFee;
+    uint256 prefixMask = type(uint256).max << (256 - 16); // 0xFFFF00..00
+    uint256 prefix = 0x0001 << (256 - 16); // 0x000100..00
+    bytes32 configDigest = bytes32((prefix & prefixMask) | (h & ~prefixMask));
+    reportContext = [configDigest, configDigest, configDigest];
 
-    // Return prank to Owner
-    vm.stopPrank();
-    vm.startPrank(OWNER_ADDRESS);
+    return (report, reportContext);
   }
-}
-
-contract FunctionsMultipleFulfillmentsSetup is FunctionsFulfillmentSetup {
-  bytes32 s_requestId2;
-  FunctionsResponse.Commitment s_requestCommitment2;
-  bytes32 s_requestId3;
-  FunctionsResponse.Commitment s_requestCommitment3;
-
-  function setUp() public virtual override {
-    FunctionsFulfillmentSetup.setUp();
 
-    // Make 2 additional requests (1 already complete)
-
-    //  *** Request #2 ***
-    vm.recordLogs();
-    s_requestId2 = s_functionsClient.sendRequest(
-      s_donId,
-      "return 'hello world';",
-      new bytes(0),
-      new string[](0),
-      new bytes[](0),
-      s_subscriptionId,
-      5500
-    );
+  function _signReport(
+    bytes memory report,
+    bytes32[3] memory reportContext,
+    uint256[] memory signerPrivateKeys
+  ) internal pure returns (bytes32[] memory rawRs, bytes32[] memory rawSs, bytes32 rawVs) {
+    bytes32[] memory rs = new bytes32[](signerPrivateKeys.length);
+    bytes32[] memory ss = new bytes32[](signerPrivateKeys.length);
+    bytes memory vs = new bytes(signerPrivateKeys.length);
+
+    bytes32 reportDigest = keccak256(abi.encodePacked(keccak256(report), reportContext));
+
+    for (uint256 i = 0; i < signerPrivateKeys.length; i++) {
+      (uint8 v, bytes32 r, bytes32 s) = vm.sign(signerPrivateKeys[i], reportDigest);
+      rs[i] = r;
+      ss[i] = s;
+      vs[i] = bytes1(v - 27);
+    }
+
+    return (rs, ss, bytes32(vs));
+  }
 
-    // Get commitment data from OracleRequest event log
-    Vm.Log[] memory entriesAfterRequest2 = vm.getRecordedLogs();
-    (, , , , , , , FunctionsResponse.Commitment memory commitment2) = abi.decode(
-      entriesAfterRequest2[0].data,
-      (address, uint64, address, bytes, uint16, bytes32, uint64, FunctionsResponse.Commitment)
+  function _reportAndStore(
+    uint256[] memory requestNumberKeys,
```

Comment:
What is requestNumberKeys?

Reply:
That's just the internal key for where to find the request in the test environment state

Reply:
I could add natspec docs to the helpers if that would help